### PR TITLE
Refactor loader

### LINKED ISSUE #452: Change `require.ensure()` to use `import()` (new webpack 2 feature)
This makes the code follow the established standards a bit more.

### PR DESCRIPTION
**Pull request by resir014**
Fixes #452.

* Upgraded babel to make dynamic imports work
* Upgraded Webpack to make sure the new chunk naming method w/ dynamic imports work
* Rewritten `require.ensure()` to `import()`
* Refactor bootstrap scripts to be more ES6-friendly

### CODE REVIEW

File: src/boot/loader.js [hunk 1]
```diff
@@ -1,65 +1,41 @@
 // This file holds the logic to generate a code-splitting loader function.
 // The code is compiled on build-time.
-var modules = {
+const modules = {
   // >> boot/modes
   // app
   //   The main game application. This will bring up the Title Screen.
-  app: 'bemuse/app',
+  app: import(/* webpackChunkName: 'app' */ 'bemuse/app'),
```
Comment (dtinth) [inline src/boot/loader.js]: `import()` is eager: The chunk starts loading when `import()` is called, not when the promise is `.then()`. This causes all modes to be downloaded right away:

image

To fix this, we can put the `import()` call into an arrow function:

```js
  app: () => import(/* webpackChunkName: 'app' */ 'bemuse/app'),
```

**Comment by resir014** [comment]
@dtinth There's a little gotcha that I still need to resolve where it sometimes complains that `regeneratorRuntime` isn't defined during runtime, and won't load. At the moment the solution is to clear the service worker cache.

image

File: src/boot/loader.js (hunk 1 above)
Comment (resir014) [reply src/boot/loader.js]: Got it, fixing this before I go to bed as well. :ok_hand:

**Comment by dtinth** [comment]
p.s. it should be safe to remove `bundle-loader` from package.json now 😄

**Comment by resir014** [comment]
@dtinth Alright, done - this should be good to merge now.

Thanks!

**Comment by dtinth** [comment]
Thanks! I’m merging this now~

**Pull request merged**